clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4605,6 +4609,56 @@ void handleFailure(Throwable throwable) {
         return new UpdateFeaturesResult(new HashMap<>(updateFutures));
     }
 
+    private Call getDecommissionBrokerCall(DecommissionBrokerOperationContext<Void> context) {
+        return new Call("decommissionBroker", context.deadline(),
+                new LeastLoadedNodeProvider()) {
+
+            @Override
+            DecommissionBrokerRequest.Builder createRequest(int timeoutMs) {
+                DecommissionBrokerRequestData data =
+                        new DecommissionBrokerRequestData().setBrokerId(context.brokerId());
+                return new DecommissionBrokerRequest.Builder(data);
+            }
+
+            @Override
+            void handleResponse(AbstractResponse abstractResponse) {
+                final DecommissionBrokerResponse response =
+                        (DecommissionBrokerResponse) abstractResponse;
+                Errors error = Errors.forCode(response.data().errorCode());
+                switch (error) {
+                    case NONE:
+                        context.future().complete(null);
+                        break;
+                    case REQUEST_TIMED_OUT:
+                        Call nextCall = getDecommissionBrokerCall(context);
```

Comment:
I think you can just raise the exception? It will get caught and `Call.fail` will be called. This does all the `tries` bookkeeping and will resubmit it if the exception is retriable (which `TimeoutException` is).

Reply:
Ah. That's much easier :)